tools/apiview/parsers/apiview-treestyle-parser-schema/models.tsp

```diff
@@ -0,0 +1,150 @@
+model CodeFile {
+  /** The version of the parser */
+  VersionString: string;
+
+  /** Name the ocde file. Usually just the package name and version*/
+  Name: string;
+  
+  Language: string;
+  LanguageVariant: "Default" | "Spring" | "Android";
+  PackageName: string;
+  ServiceName: string;
+  PackageVersion: string;
+
+  /** Maps related packages across Languages */
+  CrossLanguagePackageId: string;
+
+  /** The API Tree(s) */
+  APIForest: APITreeNode[];
+
+  Diagnostics: CodeDiagnostic[];
+}
+
+model CodeDiagnostic {
+  DiagnosticId: string;
+  Text: string;
+  HelpLinkUri: string;
+
+  /** Maps to the APITreeNode id */
+  TargetId: string;
```

Comment:
Then please delcare APITreeNodeId as a type and link it here as @lmazuel  suggested